### PR TITLE
Review and add new linters to improve the code quality of telemetry manager.

### PR DESCRIPTION
**Pull request by ameteiko**
**Configuration for Go code linting**

Changes proposed in this pull request:

- Configuration for existing linters
- Added 12 new linters
- Refactoring for the codebase to eliminate linting errors
- Makefile target for automatic code fixing

**Related issue(s)**
https://github.com/kyma-project/telemetry-manager/issues/86

### CODE REVIEW

**Comment by ameteiko** [comment]
[PR with passing integration tests](https://github.com/kyma-project/kyma/pull/17220)

File: docs/linter.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Sourcecode linting
+
+We use [golangci-lint](https://golangci-lint.run) with fine-grained configuration for the source code linting.
+
+## Linters in action
+
+Below is the list of linters configured and integrated as a CI stage through a [ProwJob](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma/components/kyma-components-static-checks.yaml#L6).
```
Comment (NHingerl) [inline docs/linter.md:7]:
```suggestion
The following linters are configured and integrated as a CI stage through a [ProwJob](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma/components/kyma-components-static-checks.yaml#L6).
```

File: docs/linter.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+| [`errchkjson`](https://github.com/breml/errchkjson) [⛭](https://golangci-lint.run/usage/linters/#errchkjson)                                                      | checks types passed to the json encoding functions               | [inline //nolint](#inline-nolint) |
+| `exportloopref`                                                                                                                                                   | finds exporting pointers for loop variables                      | [inline //nolint](#inline-nolint) |
+| [`gci`](https://github.com/daixiang0/gci) [⛭](https://golangci-lint.run/usage/linters/#gci)                                                                       | checks import order and ensures it is always deterministic       | [inline //nolint](#inline-nolint) |
+| [`ginkgolinter`](https://github.com/nunnatsa/ginkgolinter) [⛭](https://golangci-lint.run/usage/linters/#ginkgolinter)                                             | enforces standards of using ginkgo and gomega                    | [inline //nolint](#inline-nolint) |
```
Comment (NHingerl) [inline docs/linter.md:21]:
```suggestion
| [`ginkgolinter`](https://github.com/nunnatsa/ginkgolinter) [⛭](https://golangci-lint.run/usage/linters/#ginkgolinter)                                             | enforces standards of using Ginkgo and Gomega                    | [inline //nolint](#inline-nolint) |
```

**Review by NHingerl — COMMENTED** [review]
came across this PR.

Excellent job, almost no suggestions from me :D

**Review by NHingerl — APPROVED** [review]

**Pull request merged**